hq_superset/views.py

```diff
@@ -108,9 +108,40 @@ class CCHQApiException(Exception):
 
 
 def convert_to_array(string_array):
-    array_values = ast.literal_eval(string_array)
-    if array_values and array_values[0] is None:
+    """
+    Converts the string representation of a list to a list.
+    >>> convert_to_array("['hello', 'world']")
+    ['hello', 'world']
+
+    >>> convert_to_array("'hello', 'world'")
+    ['hello', 'world']
+
+    >>> convert_to_array("[None]")
+    []
+
+    >>> convert_to_array("hello, world")
+    []
+    """
+
+    def array_is_falsy(array_values):
+        if not array_values:
+            return True
+        if array_values[0] is None:
+            return True
+        return False
```

Comment:
Careful:
```python
>>> array_is_falsy([None, 1, 2, 3])
True
```

Reply:
Yes, I also thought of that scenario (and maybe should have included it in the docstrings to make it clear that it's expected behavior) and I still think it's true that it's a falsy array if there's a None at the start (or any other index...I should probably cater for that then too), since we really want an array with only real items to pass. Maybe I should rename the function to something like "array_contains_none_values", then it can become

```
def array_contains_none_values(array):
    if not array:
        return True
    if None in array:
        return None
    return False
``` 

That said, I'm open to suggestions.

Reply:
I can't make up my mind about the previous comment, so I'll put it out here.

The motivation behind the function is to not store an array as `[None]` (as would be the case for datasource expressions resulting in None that gets exported to CSV as "[None]"), but rather `[]`. Now, I don't know if I should be making the assumption that, if an array contains _any_ None values, that that array is falsy. If I decide not to make that assumption I'd have to do something like
```
if len(array) == 1 and array[0] is None:
    return True
``` 

Maybe that's better?

Reply:
Actually, now that I think of it I prefer to allow for None values being nested between real values, since I don't think I should be making assumptions about how array type datasource indicator expressions are being evaluated in HQ, but I know [None] should be [].

Reply:
Is it acceptable for a `None` value to be stored? Would it be acceptable to just remove any `None` value and return the new array?

Reply:
> The motivation behind the function is to not store an array as [None] (as would be the case for datasource expressions resulting in None that gets exported to CSV as "[None]")

I think this is crucial context. It makes sense that `[]` and `[None]` should be treated as an empty list.

In that case, what about
```python
def array_is_falsy(array_values):
    """
    Returns ``True`` if ``array_values`` is empty or ``[None]``, which
    occurs when ``None`` is exported to CSV. Otherwise returns
    ``False``.

    >>> array_is_falsy([None])
    True

    """
    return not array_values or array_values == [None]
```

Reply:
(Sorry. I should have kept scrolling. I didn't notice the commit you added 7 hours ago. :facepalm: )

Reply:
I like the one-liner!